src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -574,13 +574,25 @@ const int max_method_code_size = 64*K - 1;  // JVM spec, 2nd ed. section 4.8.1 (
 
 //----------------------------------------------------------------------------------------------------
 // old CDS options
+#if INCLUDE_CDS
 extern bool DumpSharedSpaces;
 extern bool DynamicDumpSharedSpaces;
 extern bool RequireSharedSpaces;
+#else
+extern const bool DumpSharedSpaces;
+extern const bool DynamicDumpSharedSpaces;
+extern const bool RequireSharedSpaces;
+#endif
+
+#if INCLUDE_CDS
 extern "C" {
 // Make sure UseSharedSpaces is accessible to the serviceability agent.
 extern JNIEXPORT jboolean UseSharedSpaces;
 }
+#else
+// in non CDS mode do not export it
+extern const bool UseSharedSpaces;
```

Comment:
How about this?

```
const bool RequireSharedSpaces = false;
```

This will allow the C++ compiler to eliminate all code that depend on this value when `INCLUDE_CDS` is false.

Reply:
Hi Ioi, I set this already some lines above (in the non-INCLUDE_CDS case).

Reply:
You need to remove the `extern` keyword, and add the definition of `= false` in globalDefinitions.hpp.

Then, the corresponding definition should be removed from globalDefinitions.cpp.

Reply:
Hi Ioi, I adjusted the *SharedSpaces in  globalDefinitions.cpp / globalDefinitions.hpp  .

Reply:
I am on vacation now so I won’t be able to follow up with the review. Please feel free to integrate after you get enough reviews.

Just a general comment. For blocks that are enclosed with “if (UseSharedSpace) {…}”, there’s no need to put them inside #if as these blocks will be elided by the C++ compiler. That way the code will be less messy.

Reply:
Hi Ioi, thanks for the feedback.  Mostly you are correct, the `#if`  blocks are not needed any more at most places. Should I remove them at least from my patch now or keep them?  But I think there are a few exceptions - at some places the coding attempts to set the *SharedSpaces flags and this does not compile any more after switching to const.  So there the  `#if`  blocks have to stay.